### PR TITLE
Load individual folder

### PR DESCRIPTION
**Pull request by crashbell**
I had to modify view.tree.js by adding a dummy function to work around with this lazy load. Still don't find any good way to make this bad code better. 

```javascript
TreeView.prototype.fetchData = function(opts, success) {
  // Dummy func
  throw "It must be implemented from caller when initializing this object"
}
```

### CODE REVIEW

File: src/view.tree.js [hunk 1]
```diff
@@ -99,19 +113,53 @@ TreeView.prototype.show = function(repo, treeData) {
...
-  tree.deselect_all()
-
-  // e.g. converts /buunguyen/octotree/type/branch/path to path
-  var match = path.match(/(?:[^\/]+\/){4}(.*)/)
-    , nodeId
-  if (match) {
-    nodeId = PREFIX + decodeURIComponent(match[1])
-    tree.select_node(nodeId)
-    tree.open_node(nodeId)
+
```
Comment (crashbell) [inline src/view.tree.js]: The code below is for loading nodes continuously until getting to selected node in GitHub URL.

**Comment by buunguyen** [comment]
@crashbell this is awesome! Thank you. I'll be reviewing behaviors/code and commenting here. 

First issue: enable the lazy load option, once the tree is loaded, enable the collapse folder option and save => the spinner will spin forever as there's an error. Note that updates to collapse-folder option or token will trigger a `tryLoadRepo(force=true)`.

**Comment by buunguyen** [comment]
Second: can you rebase and reissue a PR with a single commit? Much easier to review that way.

**Comment by buunguyen** [comment]
In `octotree#optionsChanged()`, you should set `reload` to true whenever the lazy-load option is changed, so that it has immediate effect.

**Comment by crashbell** [comment]
Thanks for reviewing this. I will make a single commit and revise my code regarding your feedback.

**Pull request closed without merging**